src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -158,7 +158,9 @@ def login(cmd, username=None, password=None, tenant=None, scopes=None, allow_no_
 
     if service_principal:
         from azure.cli.core.auth.identity import ServicePrincipalAuth
-        password = ServicePrincipalAuth.build_credential(password, client_assertion, use_cert_sn_issuer)
```

Comment:
Passing keyword arguments as positional arguments is fragile and may break unexpectedly.

Reply:
Agree, I am not a big fan of positional arguments in general.